### ISSUE TITLE
Don't reject on bad phaseStartTime

### ISSUE THREAD

**Issue by hannahhoward** [filecoin-project/lassie-event-recorder#24]
# What

---

in the logs we have a bunch of

---

{"level":"warn","ts":"2023-03-07T03:42:40.584Z","logger":"lassie/eventrecorder","caller":"eventrecorder/recorder.go:101","msg":"Rejected bad request with invalid event: property phaseStartTime cannot be in the future","method":"POST","path":"/v1/retrieval-events"}

---

I feel like we should make the limit more lax, given timezones and unknown operators and all. Maybe up to 1 day ahead of time.Now?

**Comment by rvagg** [comment]
Weird, you'd think operators would have their clocks synchronized properly in 2023; parsing these should be timezone independent (`Z` in this one giving that away) so this is operators reporting actual future times (I believe?). Unless we are somehow stripping timezone from these on the way out of lassie on the operator end, making it dependent on the timezone they have set, is that possible?

1 day seems like a very bit margin of error for poorly synchronized clocks.

**Comment by kylehuntsman** [comment]
Okay, so some of these validations were not in the original event recorder API, including this one. I reviewed the PR where they were added and thought sure why not. It's possible the calculation is comparing UTC against a Timezone offset of less than 0.

At the end of the day, we only care about the times relative to each other. If we can be confident that for a given retrieval_id all times will be relative to each other, than honestly they could give any datetime they want. Our eventual caclulations for ttfb and such will be accurate.
